services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/AbstractPushMonitor.java

```diff
@@ -659,6 +664,22 @@ protected void onPartitionStatusChange(OfflinePushStatus offlinePushStatus) {
     checkWhetherToStartBufferReplayForHybrid(offlinePushStatus);
   }
 
+  private boolean disableReplica(PartitionAssignment partitionAssignment, Optional<String> leaderErrorInfo) {
+    String errorInfo = leaderErrorInfo.orElse("");
```

Comment:
Is there a more explicit way to determine if the ERROR replica is a leader replica? Worried that in the future when we change the behavior for whether to populate the status messages for leader replicas that this logic here will just break silently.

Reply:
I can repopulate the info again without passing secret messages here. Since the error case is infrequent, it should be fine?